### PR TITLE
fix(data): add selective dependency resolution for xml2js

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes

Fixes https://github.com/aws-amplify/amplify-js/security/dependabot/63

### [Successful pipeline run against E2E tests](https://github.com/aws-amplify/amplify-js/actions/runs/8162180554/job/22312801117)

Note: one of the E2E tests is failing, but this is due to an expired API key and unrelated to this change.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/push-preid-release.yml
Comment (iartemiev) [inline .github/workflows/push-preid-release.yml:1]: This change should be reverted before merging, right?

File: .github/workflows/push-preid-release.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
   push:
     branches:
       # Change this to your branch name where "example-preid" corresponds to the preid you want your changes released on
-      - feat/example-preid-branch/main
+      - dependabot-63-fix
```
Comment (jimblanc) [inline .github/workflows/push-preid-release.yml:12]: Echoing @iartemieve, these changes need to be reverted. Otherwise looks good though

**Review by jimblanc — CHANGES_REQUESTED** [review]

**Review by iartemiev — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by stocaaro — APPROVED** [review]

**Review by jimblanc — CHANGES_REQUESTED** [review]
Need to update lockfile as well

File: .github/workflows/push-preid-release.yml (hunk 1 above)
Comment (david-mcafee) [reply .github/workflows/push-preid-release.yml:12]: Had to move the resolution to the top-level `package.json`, so I've temporarily added this back to test in the pipeline. Will let you know when this is ready for re-review

File: .github/workflows/push-preid-release.yml [hunk 2]
```diff
@@ -25,14 +25,13 @@ jobs:
...
-    needs:
-      - e2e
-      - parse-preid
-    secrets: inherit
-    uses: ./.github/workflows/callable-npm-publish-preid.yml
-    # The preid should be detected from the branch name recommending feat/{PREID}/whatever as branch naming pattern
-    #   if your branch doesn't follow this pattern, you can override it here for your branch.
-    with:
-      preid: ${{ needs.parse-preid.outputs.preid }}
+#   preid-release:
```
Comment (david-mcafee) [inline .github/workflows/push-preid-release.yml:28]: Same here

**Review by iartemiev — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**